### PR TITLE
agent store: accept internal ts and return internal id

### PR DESCRIPTION
**Pull request by efim-poberezkin**

### CODE REVIEW

File: src/Simplex/Messaging/Agent.hs [hunk 1]
```diff
@@ -164,10 +164,11 @@ processCommand c@AgentClient {sndQ} (corrId, connAlias, cmd) =
         _ -> throwError PROHIBITED -- NOT_READY ?
       where
         sendMsg sq = do
-          withStore $ \st -> createSndMsg st connAlias msgBody
+          senderTs <- liftIO getCurrentTime
+          senderId <- withStore $ \st -> createSndMsg st connAlias msgBody senderTs
           sendAgentMessage c sq $ A_MSG msgBody
-          -- TODO respond $ SENT aMsgId
-          respond $ SENT 0
+          -- ? change AgentMsgId to Int?
```
Comment (epoberezkin) [inline src/Simplex/Messaging/Agent.hs:170]: Int64? yes

File: src/Simplex/Messaging/Agent.hs [hunk 2]
```diff
@@ -164,10 +164,11 @@ processCommand c@AgentClient {sndQ} (corrId, connAlias, cmd) =
         _ -> throwError PROHIBITED -- NOT_READY ?
       where
         sendMsg sq = do
-          withStore $ \st -> createSndMsg st connAlias msgBody
+          senderTs <- liftIO getCurrentTime
```
Comment (epoberezkin) [inline src/Simplex/Messaging/Agent.hs:167]: this timestamp should be the same as timestamp used in `mkAgentMessage` - it should be passed there, so they are the same

Comment (epoberezkin) [reply src/Simplex/Messaging/Agent.hs:167]: and it should also be the same as internalTs I think for sent messages

File: src/Simplex/Messaging/Agent.hs [hunk 3]
```diff
@@ -250,17 +251,22 @@ processSMPTransmission c@AgentClient {sndQ} (srv, rId, cmd) = do
...
+              recipientId <- withStore $ \st ->
+                createRcvMsg
+                  st
+                  connAlias
+                  body
+                  recipientTs
+                  senderMsgId
+                  senderTimestamp
+                  srvMsgId
+                  srvTs
```
Comment (epoberezkin) [inline src/Simplex/Messaging/Agent.hs:264]: a single long line can be easier to read

File: src/Simplex/Messaging/Agent.hs [hunk 4]
```diff
@@ -250,17 +251,22 @@ processSMPTransmission c@AgentClient {sndQ} (srv, rId, cmd) = do
...
               notify connAlias $
                 MSG
                   { m_status = MsgOk,
-                    m_recipient = (0, recipientTs),
+                    -- ? change AgentMsgId to Int?
```
Comment (epoberezkin) [inline src/Simplex/Messaging/Agent.hs:268]: Int64

File: src/Simplex/Messaging/Agent.hs (hunk 3 above)
Comment (efim-poberezkin) [reply src/Simplex/Messaging/Agent.hs:264]: agreed.. it just goes beyond 120 length which becomes a bit too much.. for now w/e I guess, we have some signatures in Store that also exceed 120. Will return to it once we set up line length rules

**Review by epoberezkin — APPROVED** [review]

**Pull request merged**